llama_index/callbacks/finetuning_handler.py

```diff
@@ -0,0 +1,103 @@
+import json
+from typing import Any, Dict, List, Optional
+
+from llama_index.callbacks.base import BaseCallbackHandler
+from llama_index.callbacks.schema import CBEventType, EventPayload
+
+
+class OpenAIFineTuningHandler(BaseCallbackHandler):
+    def __init__(
+        self,
+    ) -> None:
+        """Initialize the base callback handler."""
+        super().__init__(
+            event_starts_to_ignore=[],
+            event_ends_to_ignore=[],
+        )
+        self._finetuning_events: Dict[str, List[Any]] = {}
+
+    def on_event_start(
+        self,
+        event_type: CBEventType,
+        payload: Optional[Dict[str, Any]] = None,
+        event_id: str = "",
+        **kwargs: Any,
+    ) -> str:
+        """Run when an event starts and return id of event."""
+        from llama_index.llms.base import ChatMessage, MessageRole
```

Comment:
is this because of circular imports

Reply:
(not in this PR) in the future we could prob move struct types that have light dependencies into `llama_index.schemas` or something

Reply:
Yea it's circular imports 😢